impl/client/src/main/javascript/web/pentaho/visual/base/view.js

```diff
@@ -0,0 +1,1322 @@
+       * [update]{@link pentaho.visual.base.View#update}.
+       * This allows the _container application_ to further configure the view,
+       * like, for example,
+       * setting the [isAutoUpdate]{@link pentaho.visual.base.View#isAutoUpdate} property
+       * or registering event listeners,
+       * before the initial update.
+       *
+       * Over time, the view's model is mutated and, in response,
+       * the view detects these changes, marks itself [dirty]{@link pentaho.visual.base.View#isDirty},
+       * and, by default, automatically updates itself.
+       *
+       * When a view is no longer needed,
+       * the _container application_ **must** call its [dispose]{@link pentaho.visual.base.View#dispose} method,
+       * so that the view can free held _resources_ and not cause memory-leaks.
+       *
+       * @constructor
+       * @description Constructs a `View` instance.
+       * @param {pentaho.visual.base.spec.IViewEx} [viewSpec] - The extended view specification.
+       *
+       * @see pentaho.visual.base.spec.IView
+       * @see pentaho.visual.base.spec.IViewType
+       */
+      constructor: function(viewSpec) {
+
+        this.base(viewSpec);
+
+        /**
+         * The container element where the view is rendered.
+         *
+         * @type {DOMElement}
+         * @readOnly
+         * @private
+         */
+        this.__domContainer = null;
+
+        /**
+         * The promise for the completion of the current update operation, if any; or `null`.
+         *
+         * @type {Promise}
+         * @private
+         */
+        this.__updatingPromise = null;
+
+        /**
+         * Indicates if the view is automatically updated whenever the model is changed.
+         *
+         * @type {boolean}
+         * @default true
+         * @private
+         */
+        this.__isAutoUpdate = true;
+
+        /**
+         * The set of dirty property groups of the view.
+         *
+         * @type {!pentaho.util.BitSet}
+         * @readOnly
+         * @private
+         */
+        this.__dirtyPropGroups = new BitSet(View.PropertyGroups.All); // mark view as initially dirty
+
+        // Initialize any special properties provided directly in viewSpec.
+        if(viewSpec) {
+          if(viewSpec.domContainer) this.domContainer = viewSpec.domContainer;
+          if(viewSpec.isAutoUpdate != null) this.isAutoUpdate = viewSpec.isAutoUpdate;
+        }
+
+        // Let any mixins initialize.
+        this._init(viewSpec);
+      },
+
+      /**
+       * Called when the view is constructed.
+       *
+       * Override this method, from a configuration mixin class,
+       * to perform any initialization.
+       *
+       * @param {pentaho.visual.base.spec.IViewEx} viewSpec - The view specification provided at construction time,
+       * if any.
+       * @protected
+       */
+      _init: function(viewSpec) {
```

Comment:
The actual, more perfect, convention, that we have failed to apply consistently, is to use `__` for private and `_` for protected. This is one of the few classes where we have done this consistently. Some `pentaho/type` classes also do this consistently. I'd love a story to fix this.